chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -255,10 +257,14 @@ async def add_coin(
         for coin_info in my_nft_coins:
             if coin_info.coin == coin:
                 my_nft_coins.remove(coin_info)
+                delete_off_chain_metadata(coin_info.nft_id, self.wallet_state_manager.config.get(CACHE_PATH_KEY, None))
         new_nft = NFTCoinInfo(nft_id, coin, lineage_proof, puzzle, mint_height, confirmed_height)
         my_nft_coins.append(new_nft)
         await self.wallet_state_manager.nft_store.save_nft(self.id(), self.get_did(), new_nft)
         await self.wallet_state_manager.add_interested_coin_ids([coin.name()])
+        multiprocessing.Process(
```

Comment:
if user receives a lot of new coins this will fork same amount of processes and given fetching metadata is mainly I/O bound an `asyncio.create_task` should be sufficient?
another potential issue here is that if we receive 1000 NFTs, we would issue 1000 requests crushing users bandwidth/connection, so should be rate limited

Reply:
I would rather not spawn lots of tasks then block them all on a semaphore or such so they sit around doing nothing.  But, that's what we do now and I haven't gotten my reference async worker pool into use so go for it for right now.  The idea being to spawn as many workers as you want working and have them just read from a queue of work to be done.

Reply:
Are you referring thread pooling?

Reply:
No, asyncio tasks that just process work from a queue.  But sure, same idea as a pool of threads or processes.  I have an old PR I need to resurrect to give us a reusable form for this.

Reply:
I think we don't have to create another thread for loading metadata. This thread is only for this coin right?

Reply:
I tried asyncio, but it doesn't work here because of it is in an async event loop.